consensus/service/src/byzantine_ledger/worker.rs

```diff
@@ -87,6 +91,11 @@ pub struct ByzantineLedgerWorker<
 
     // Ledger sync state.
     ledger_sync_state: LedgerSyncState,
+
+    // A map of responder id to a list of tx hashes that it is unable to provide. This allows us to
+    // skip attemping to fetch txs that are bound to fail. A BTreeSet is used to speed up lookups
+    // as expect to be doing more lookups than inserts.
+    missing_tx_hashes: HashMap<ResponderId, BTreeSet<TxHash>>,
```

Comment:
This name is a little confusing because elsewhere in this file we use "missing_hashes" to refer to tx_hashes that are not in TxManager, but here "missing" means we failed to download the transaction.
Maybe give this a different name like `unavailable_tx_hashes` or `failed_tx_requests`?